### PR TITLE
1) fix small buffer for 16k wpp stream;  2)  clear vector when re-allocate buffer

### PR DESCRIPTION
**Pull request by weizhu-intel**
There two changes are all cherry-picked from internal branch.
1) For 16k wpp stream,  it has up to 300 bytes' prevention, so increase 300 bytes here. Meanwhile, 300 bytes is the  biggest size we met till now.

### CODE REVIEW

**Comment by dmitryermilov** [comment]
@weizhu-intel , please give a better name to this PR

File: _studio/shared/umc/codec/h265_dec/src/umc_h265_va_supplier.cpp [hunk 1]
```diff
@@ -174,7 +174,7 @@ UMC::Status VATaskSupplier::AllocateFrameData(H265DecoderFrame * pFrame, mfxSize
 H265Slice * VATaskSupplier::DecodeSliceHeader(UMC::MediaDataEx *nalUnit)
 {
     size_t dataSize = nalUnit->GetDataSize();
-    nalUnit->SetDataSize(std::min<size_t>(2560, dataSize));
+    nalUnit->SetDataSize(std::min<size_t>(2860, dataSize));
```
Comment (dvrogozh) [inline _studio/shared/umc/codec/h265_dec/src/umc_h265_va_supplier.cpp:177]: This is duplicated change with #2092. Please, left a single PR for review.

**Comment by eponomar** [comment]
Currently I don't see any issues in CI tests after last force-push.

File: _studio/shared/umc/codec/h265_dec/src/umc_h265_va_supplier.cpp (hunk 1 above)
Comment (dvrogozh) [inline _studio/shared/umc/codec/h265_dec/src/umc_h265_va_supplier.cpp:177]: why this is 2860, not 2861 or 2863? Can you, please, create a macro def or constant with the self-speaking name explaining the value. If 2860 is composite value - explain how it was constructed.

Comment (weizhu-intel) [reply _studio/shared/umc/codec/h265_dec/src/umc_h265_va_supplier.cpp:177]: Thanks Dmitry V. Pls review agian.

BTW,  will this comment be public to open source? Or just internal comment?
If public, is there any format/rule to comply will?

Comment (dvrogozh) [reply _studio/shared/umc/codec/h265_dec/src/umc_h265_va_supplier.cpp:177]: This is open source project and this should be a public comment. Basically, not explaining what this magic value actually means and/or where it comes from undermines open source status of the project.

**Pull request merged**